tests/git.zunit

```diff
@@ -1355,6 +1355,632 @@
     _fzf_complete_git 'git add '
 }
 
+<<<<<<< HEAD
+@test 'Testing completion: git pull --recurse-submodules=**' {
+    _fzf_complete() {
+        assert $# equals 2
+        assert $1 same_as '--ansi --tiebreak=index '
+        assert $2 same_as 'git pull '
+
+        run cat
+        assert ${#lines} equals 3
+        assert ${lines[1]} same_as --recurse-submodules=yes
+        assert ${lines[2]} same_as --recurse-submodules=on-demand
+        assert ${lines[3]} same_as --recurse-submodules=no
+    }
+
+    prefix=--recurse-submodules=
+    _fzf_complete_git 'git pull '
+}
+
+@test 'Testing completion: git pull --recurse-submodules **' {
```

Comment:
I don't think those tests that examine the situation where the last argument is a long option that takes an optional argument and no prefix is given are unnecessary. They don't test anything but produce confusion.